File: src/server/utils/math/numeric/Hexadecimal.ts
// Funções auxiliares com hexadecimal

import { RandInt } from "@utils/math/basics.js"

/**
 * Gera uma string hexadecimal aleatória
 * 
 * @param length Tamanho da string
 */
function GenerateHexadecimal(length: number): string {
    if (length < 1) {
        throw new Error("Hexadecimal length should be at least 1.")
    }

    let characters = [
        "0", "1", "2", "3",
        "4", "5", "6", "7",
        "8", "9", "A", "B",
        "C", "D", "E", "F"
    ]

    let hex = ""

    for (let i = 0; i < length; ++i) {
        let digit = RandInt(0, 15)
        hex += characters[digit]
    }

    return hex
}

export { GenerateHexadecimal }

File: src/client/utils/math/numeric/hexadecimal.py
# Funções para manipulação de hexadecimal

from random import randint

# Gera uma cadeia de caracteres de hexadecimal
def GenerateHexadecimalStr(length: int) -> str:
    if length < 1:
        raise ValueError("Length should be at least 1.")
    
    chars = [
        "0", "1", "2", "3",
        "4", "5", "6", "7",
        "8", "9", "A", "B",
        "C", "D", "E", "F"
    ]

    hex = ""

    for i in range(length):
        index = randint(0, 15)
        hex += chars[index]

    return hex
